isis/src/control/objs/BundleUtilities/BundleObservation.cpp

```diff
@@ -783,6 +784,785 @@ namespace Isis {
+      // instrumentPointing's center angles for RA, DEC, and TWI
+      if (nPointingCoefficients == 0) {
+        nPointingCoefficients = 1;
+        useDefaultPointing = true;
+      }
+
+      // Force number of position and pointing parameters to each be 3 (X,Y,Z; RA,DEC,TWI)
+      // so we can always output a value for them
+      int nPositionParameters = 3 * nPositionCoefficients;
+      int nPointingParameters = 3 * nPointingCoefficients;
+      int nParameters = nPositionParameters + nPointingParameters;
+
+      coefX.resize(nPositionCoefficients);
+      coefY.resize(nPositionCoefficients);
+      coefZ.resize(nPositionCoefficients);
+      coefRA.resize(nPointingCoefficients);
+      coefDEC.resize(nPointingCoefficients);
+      coefTWI.resize(nPointingCoefficients);
+
+      if (m_instrumentPosition) {
+        if (!useDefaultPosition) {
+          m_instrumentPosition->GetPolynomial(coefX, coefY, coefZ);
+        }
+        // Use the position's center coordinate if not solving for spacecraft position
+        else {
+          const std::vector<double> centerCoord = m_instrumentPosition->GetCenterCoordinate();
+          coefX[0] = centerCoord[0];
+          coefY[0] = centerCoord[1];
+          coefZ[0] = centerCoord[2];
+        }
+      }
+
+      if (m_instrumentRotation) {
+        if (!useDefaultPointing) {
+          m_instrumentRotation->GetPolynomial(coefRA, coefDEC, coefTWI);
+        }
+        // Use the pointing's center angles if not solving for pointing (rotation)
+        else {
+          const std::vector<double> centerAngles = m_instrumentRotation->GetCenterAngles();
+          coefRA[0] = centerAngles[0];
+          coefDEC[0] = centerAngles[1];
+          coefTWI[0] = centerAngles[2];
+        }
+      }
+
+      // for convenience, create vectors of parameters names and values in the correct sequence
+      std::vector<double> finalParameterValues;
+      QStringList parameterNamesList;
+      QStringList correctionUnitList;
+
+        QString str("%1(%2)  ");
+
+
+        if (nPositionCoefficients > 0) {
+          for (int i = 0; i < nPositionCoefficients; i++) {
+            finalParameterValues.push_back(coefX[i]);
+            if (i == 0) {
+              parameterNamesList.append( str.arg("  X  ").arg("km") );
+              correctionUnitList.append("m");
+            }
+            else {
+              switch(i) {
+              case 1:
+                parameterNamesList.append( str.arg("     ").arg("km/s") );
+                correctionUnitList.append("m/s");
+                break;
+              case 2:
+                parameterNamesList.append( str.arg("     ").arg("km/s^2") );
+                correctionUnitList.append("m/s^2");
+                break;
+              case 3:
+                parameterNamesList.append( str.arg("     ").arg("km/s^3") );
+                correctionUnitList.append("m/s^3");
+                break;
```

Comment:
I believe that we agreed that cases 2 and 3 could be removed as the default would handle both of these.

Reply:
See my comment above.  The bundleOutput function was the original function with the nested for-loops which was refactored as bundleOutputString.  I kept it in the code for testing purposes to compare with the refactored one but it has now been deleted.  Sorry for the confusion.